libraries/kotlinx-metadata/jvm/src/kotlinx/metadata/jvm/jvmExtensionVisitors.kt

```diff
@@ -212,6 +212,19 @@ open class JvmPropertyExtensionVisitor @JvmOverloads constructor(
         delegate?.visitSyntheticMethodForAnnotations(signature)
     }
 
+    /**
+     * Visits the JVM signature of a synthetic method which is generated when a delegated property's delegate object is
+     * optimized out, e.g. because it is constant; in that case, a copy of that object can be obtained on demand by calling
+     * this method. It takes up to two arguments - the property's receivers.
+     *
+     * Example: `JvmMethodSignature("getX$delegate", "(LMyClass;)LMyDelegate;")`
+     *
+     * @param signature the signature of the synthetic method
+     */
+    open fun visitSyntheticMethodForDelegate(signature: JvmMethodSignature?) {
```

Comment:
Please also override it in `JvmMetadataExtensions.writePropertyExtensions` and add a test on kotlinp, for example into `libraries/tools/kotlinp/testData/Properties.kt`

Reply:
Done. kotlinp tests did not use the IR backend before, though, and turning it on revealed an issue in JVM_IR metadata: the signature of `invoke` was never serialized in lambda classes (so kotlinx-metadata attempted to guess, and got it wrong). Not sure whether this affects anything in the compiler - the only place I know of where we care about `invoke`'s signature is the inliner, but there we just take whatever non-synthetic `invoke` exists, and this is all broken anyway (#4480).